product/views.py

```diff
@@ -0,0 +1,43 @@
+import json
+from random                     import uniform
+
+from django.views               import View
+from django.http                import JsonResponse
+from django.core.exceptions     import ValidationError
+
+from product.models             import Product
+
+class ProductDetailView(View):
+    def get(self ,request):
+        try:
+            pk = request.GET.get('id',None)
+
+            if pk is not None:
```

Comment:
있는 경우가 아닌 없는 경우를 먼저 판단해서 예외처리하면 조금 더 깔끔해 집니다.
```suggestion
            if Product.objects.filter(id=product_id).exists():
                return JsonResponse({'message': 'NOT_FOUND'}, status=404)
```